### PR TITLE
Fix: missing heading for intersections

### PR DESCRIPTION
**Pull request by Neyromancer**
Originally only 2 edges were added along the route after the fix any possible number of intersecting edges are added to the final route.

### CODE REVIEW

**Comment by Neyromancer** [comment]
Tets to be added

**Comment by Neyromancer** [comment]
Tests added

File: src/tyr/route_serializer_osrm.cc
Comment (merkispavel) [inline src/tyr/route_serializer_osrm.cc:553]: 1. `arrive_maneuver` is always equal to `false` in this branch so this variable/ternary operator is redundant
2. The entry is set to false if the intersecting edge is not routable in that direction. I'm not sure that is should be always false..

File: src/tyr/route_serializer_osrm.cc
Comment (merkispavel) [inline src/tyr/route_serializer_osrm.cc:556]: as these are outgoing edges - boolean flags should be set vice versa, nope?
```suggestion
        edges.emplace_back(intersecting_edge->begin_heading(), entry, false, true);
```

File: src/tyr/route_serializer_osrm.cc
Comment (merkispavel) [inline src/tyr/route_serializer_osrm.cc:552]: I would at it to the code where "main" outgoing edge is added
```cpp
// Add the edge departing the node
    if (!arrive_maneuver) {
      edges.emplace_back(curr_edge->begin_heading(), true, false, true);
      // add other outgoing edges
      ...
    }
```

**Review by merkispavel — CHANGES_REQUESTED** [review]

File: src/tyr/route_serializer_osrm.cc
Comment (Neyromancer) [reply src/tyr/route_serializer_osrm.cc:552]: the logic here is different, we are adding intersecting edges that are not starting or not ending, only in the middle of the road.

File: src/tyr/route_serializer_osrm.cc
Comment (merkispavel) [inline src/tyr/route_serializer_osrm.cc:544]: wow, what a nice method. Exactly what we need!

**Review by merkispavel — APPROVED** [review]
Nice, looks good

**Review by merkispavel — APPROVED** [review]
Re approve after merging master with conflicts

**Pull request merged**
